lib/file-source/src/fingerprinter.rs

```diff
@@ -125,6 +118,27 @@ impl Fingerprinter {
             })
             .ok()
     }
+
+    pub fn get_bytes_checksum(
+        &self,
+        path: &PathBuf,
+        buffer: &mut Vec<u8>,
+    ) -> Result<Option<FileFingerprint>, io::Error> {
+        match self.strategy {
+            FingerprintStrategy::Checksum {
+                bytes,
+                ignored_header_bytes,
+            } => {
+                buffer.resize(bytes, 0u8);
+                let mut fp = fs::File::open(path)?;
+                fp.seek(io::SeekFrom::Start(ignored_header_bytes as u64))?;
+                fp.read_exact(&mut buffer[..bytes])?;
```

Comment:
I wonder if it would be better to read what is available, and allow the trailing 0s? We could call `read()` iteratively until we had the bytes or an EOF?